### PR TITLE
Sys_core XEH_postInit cleanup

### PR DESCRIPTION
**Pull request by TheMagnetar**
**When merged this pull request will:**
- Separate two eventhandler functions into their own files: coreInitPFH and vehicleCrewPFH
- Needs testing.

### CODE REVIEW

File: addons/sys_core/XEH_postInit.sqf [hunk 1]
```diff
@@ -112,42 +94,22 @@ if (GVAR(revealToAI) && hasInterface) then {
...
-        } else {
-            ACRE_PLAYER_VEHICLE_CREW = [];
-        };
-    } else {
-        ACRE_PLAYER_VEHICLE_CREW = [];
-    };
-};
-ADDPFH(_vehicleCrewPFH, 1.1, []);
+ACRE_PLAYER_VEHICLE_CREW = [];
+ADDPFH(DFUNC(vehicleCrewPFH), 1.1, []);
```
Comment (Sniperhid) [inline addons/sys_core/XEH_postInit.sqf:99]: `DFUNC` is a macro for declaring functions, it should certainly not be used. Use `FUNC` over `DFUNC` where referencing a function for use.

**Review by Sniperhid — CHANGES_REQUESTED** [review]

File: addons/sys_core/XEH_postInit.sqf (hunk 1 above)
Comment (TheMagnetar) [reply addons/sys_core/XEH_postInit.sqf:99]: I have seen this in several other files, specially those within the ADDPFH macro, like in fnc_enableRevealAI.sqf, fnc_handleMultiPttKeyPressUp.sqf, fnc_utilityFunction.sqf in sys_core alone. I wasn't sure about them so I kept it as they were originally written. Should all of these be then changed?

File: addons/sys_core/fnc_coreInitPFH.sqf [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+[] call FUNC(muting);
+[] call FUNC(speaking);
+
+//Set the speaking volume to normal.
+[.7] call acre_api_fnc_setSelectableVoiceCurve;
+acre_sys_gui_VolumeControl_Level = 0;
+
+ACRE_CORE_INIT = true;
+TRACE_1("ACRE CORE INIT", ACRE_CORE_INIT);
+[(_this select 1)] call CBA_fnc_removePerFrameHandler;
```
Comment (jonpas) [inline addons/sys_core/fnc_coreInitPFH.sqf]: No need for parenthesis inside array.

File: addons/sys_core/fnc_coreInitPFH.sqf [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * Public: No
+ */
+#include "script_component.hpp"
+
+if (isNull player) exitWith { };
+acre_player = player;
+
+if (!ACRE_MAP_LOADED) exitWith { };
+if (!ACRE_DATA_SYNCED) exitWith { };
+if (GVAR(ts3id) == -1) exitWith { };
```
Comment (jonpas) [inline addons/sys_core/fnc_coreInitPFH.sqf]: `if (!ACRE_MAP_LOADED || {!ACRE_DATA_SYNCED} || {GVAR(ts3id) == 1}) exitWith {};`
Possibly not faster, but cleaner.

File: addons/sys_core/fnc_coreInitPFH.sqf [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ * None
+ *
+ * Example:
+ * [] call acre_sys_core_fnc_coreInitPFH
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+if (isNull player) exitWith { };
```
Comment (jonpas) [inline addons/sys_core/fnc_coreInitPFH.sqf]: No spaces in empty code blocks.

File: addons/sys_core/XEH_postInit.sqf [hunk 5]
```diff
@@ -80,26 +80,8 @@ if (!([findDisplay 0] isEqualTo allDisplays)) then {
...
-    //Set the speaking volume to normal.
-    [.7] call acre_api_fnc_setSelectableVoiceCurve;
-    acre_sys_gui_VolumeControl_Level = 0;
-
-    ACRE_CORE_INIT = true;
-    TRACE_1("ACRE CORE INIT", ACRE_CORE_INIT);
-    [(_this select 1)] call CBA_fnc_removePerFrameHandler;
-};
 
+// Check whether ACRE2 is fully loaded.
```
Comment (jonpas) [inline addons/sys_core/XEH_postInit.sqf]: No final dot.

**Review by jonpas — CHANGES_REQUESTED** [review]

**Review by jonpas — APPROVED** [review]

**Review by Sniperhid — APPROVED** [review]

**Pull request merged**